blocks/library/paragraph/index.js

```diff
@@ -425,20 +436,27 @@ export const settings = {
 			dropCap,
 			backgroundColor,
 			textColor,
+			customBackgroundColor,
+			customTextColor,
 			fontSize,
 			customFontSize,
 		} = attributes;
 
+		const textClass = getColorClass( 'text', textColor );
+		const backgroundClass = getColorClass( 'background', backgroundColor );
+
 		const className = classnames( {
 			[ `align${ width }` ]: width,
-			'has-background': backgroundColor,
+			'has-background': backgroundColor || customBackgroundColor,
 			'has-drop-cap': dropCap,
 			[ `is-${ fontSize }-text` ]: fontSize && FONT_SIZES[ fontSize ],
+			[ textClass ]: textClass,
+			[ backgroundClass ]: backgroundClass,
 		} );
 
 		const styles = {
-			backgroundColor: backgroundColor,
-			color: textColor,
+			backgroundColor: ! backgroundColor ? customBackgroundColor : undefined,
```

Comment:
I'm a bit confused here by this logic. When will `backgroundColor` be assigned as a style?

Should this be...

```js
backgroundColor: customBackgroundColor || backgroundColor,
```

Same problem on the following line.

Reply:
The logic `backgroundColor: ! backgroundColor ? customBackgroundColor : undefined`, was correct it means if we don't have a named color use the custom color. But it was not intuitive I changed to `backgroundColor: backgroundClass ? undefined : customBackgroundColor` which means don't set any style if we have a class set the background color to customBackgroundColor if we don't have a class.